src/websocket/server.cc

```diff
@@ -183,23 +267,65 @@ future<> connection::read_loop() {
         });
     }).then_wrapped([this] (future<> f) {
         if (f.failed()) {
-            wlogger.error("Read failed: {}", f.get_exception());
+            wlogger.error("Failure: {}", f.get_exception());
         }
         return _replies.push_eventually({});
     }).finally([this] {
         return _read_buf.close();
     });
 }
 
+future<> connection::send_data(temporary_buffer<char> buff) {
+    sstring data;
+    data.append("\x81", 1);
+    if ((126 <= buff.size()) && (buff.size() <= std::numeric_limits<uint16_t>::max())) {
+        uint16_t length = buff.size();
+        length = htobe16(length);
+        data.append("\x7e", 1);
+        data.append(reinterpret_cast<char*>(&length), sizeof(uint16_t));
+    } else if (std::numeric_limits<uint16_t>::max() < buff.size()) {
+        uint64_t length = buff.size();
+        length = htobe64(length);
+        data.append("\x7f", 1);
+        data.append(reinterpret_cast<char*>(&length), sizeof(uint64_t));
+    } else {
+        uint8_t length = buff.size() & 0x7F;
+        data.append(reinterpret_cast<char*>(&length), sizeof(uint8_t));
+    }
+    data.append(buff.get(), buff.size()); 
+    temporary_buffer<char> result{data.data(), data.length()}; 
+    return _write_buf.write(std::move(result)).then([this]{
+        return _write_buf.flush();
+    }); 
+}
+
 future<> connection::response_loop() {
-    // FIXME: implement
-    return make_ready_future<>();
+    return do_until([this] {return _done;}, [this] {
+        return input().read().then([this](temporary_buffer<char> buf) {
+            // FIXME: implement
```

Comment:
Would be nice to have in the commit message a summary of what's missing and what the plan is.